cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -300,6 +315,15 @@ func (csr *ClusterStateRegistry) updateScaleRequests(currentTime time.Time) {
 	csr.scaleDownRequests = newScaleDownRequests
 }
 
+func (csr *ClusterStateRegistry) allInstancesAreCreated(nodeGroupId string) bool {
```

Comment:
Firstly, this should be called `areAllNodeCreated`. Secondly, this does not use passed `nodeGroupId`, which makes me think that it is not working as intended. Maybe with a single ongoing scale-up it works as expected, but I totally don't understand how this will behave if there are multiple ongoing scale0ups at once, especially if those would be in the same node group.

Reply:
Yeah, thanks for spotting this. Added filtering by nodeGroupId and extended TestScaleUpNodeRegistrationTimeout test so that reversing this change will fail an unit test.